invenio_rdm_records/records/systemfields/access/field.py

```diff
@@ -60,6 +65,10 @@ def __init__(
         if grants is None:
             self._grants = grants_cls()
 
+        self._links = links
+        if links is None:
+            self._links = links_cls()
```

Comment:
i changed them to `self._x = x if x is not None else x_cls()`; that still avoids the falsy empty sets, and is more concise

Reply:
oh, `_links` can hold a set or a "Links class-like" that doesn't behave like a set? or at least doesn't have the `__bool__` of a set?  It's merged now, so we can see later...

Reply:
oh no, i was thinking that as soon as the user explicitly specified anything other than `None` for the `links` argument in the constructor, then that should be used (the default value *is* `None` for all the arguments, causing the fallback to the configured default classes)
in that case, the default semantics of `__bool__` for sets *are* the problem, because empty sets evaluate to `False` -- just like `bool(None)`, causing us to lose the ability to distinguish between the two cases
but what if the user wanted to specify their own thing, and it just happens to be empty?

the primary background of this thought is to give users of this API the possibility of easily overriding the default behavior
a la "not satisfied with the default `Links` list implementation that we supply? just pass something to the constructor that's more to your liking"
though thinking about it, that's probably never gonna be required anyway? :sweat_smile:

Reply:
> but what if the user wanted to specify their own thing, and it just happens to be empty?

then wouldn't that be equivalent to the default `links_cls()` anyway ? ... but really the below is the approach for me:

>  though thinking about it, that's probably never gonna be required anyway?

That's my thinking. I am definitely conservative on that front though (and maybe others have differing opinion): you ain't gonna need it yet and all that :smiley: .

Reply:
> then wouldn't that be equivalent to the default links_cls() anyway ? ... but really the below is the approach for me:

not if it's an instance of another custom class, different from the supplied `links_cls` :stuck_out_tongue: 
but yeah, let's simplify.